lightning-c-bindings/README.md

```diff
@@ -0,0 +1,79 @@
+The wrapper crate and C/C++ Headers in this folder are auto-generated from the Rust-Lightning
+source by the c-bindings-gen crate contained in the source tree and
+[cbindgen](https://github.com/eqrion/cbindgen). They are intended to be used as a base for building
+language-specific bindings, and are thus incredibly low-level and may be difficult to work with
+directly.
+
+LDK C Bindings
+===================
+
+The C bindings available at include/lightning.h require inclusion of include/rust_types.h first.
+
+All of the Rust-Lightning types are mapped into C equivalents which take a few forms, namely:
+
+ * Structs are mapped into a simple wrapper containing a pointer to the native Rust-Lightning
+   object and a flag to indicate whether the object is owned or only a reference. Such mappings
+   usually generate a X_free function which must be called to release the allocated resources.
+   Note that calling X_free will do nothing if the underlying pointer is NULL or if the reference
+   flag is set.
+
+   You MUST NOT create such wrapper structs manually, relying instead on constructors which have
+   been mapped from equivalent Rust constructors.
+
+ * Traits are mapped into a concrete struct containing a void pointer for your use and a jump table
+   listing the functions which the trait must implement. The void pointer is free for use as you
+   see fit and passed as the first argument to all function calls, but note that no destructor
+   function is (currently) provided.
+
+   Rust native structs which implement a trait result in the generation of an X_as_Y function which
+   allows you to use the native Rust object in place of the trait. Such generated objects are only
+   valid as long as the original Rust native object is owned by a C-wrapped struct, and has not been
```

Comment:
You might imagine a different bindings implementation which does some kind of reference counting to prevent null pointer dereferences? I'm not sure exactly which clarification you want here.